azkaban-web-server/src/test/java/azkaban/scheduler/QuartzSchedulerTest.java

```diff
@@ -87,74 +85,60 @@ public void init() {
     assertThatThrownBy(
-        () -> scheduler.registerJob("0 5 * * * *", createJobDescription()))
+        () -> scheduler.scheduleIfAbsent("0 5 * * * *", createJobDescription()))
         .isInstanceOf(SchedulerException.class)
         .hasMessageContaining("The cron expression string");
   }
 
   @Test
-  public void testUnregisterSchedule() throws Exception {
-    scheduler.registerJob("* * * * * ?", createJobDescription());
+  public void testUnschedule() throws Exception {
+    scheduler.scheduleIfAbsent("* * * * * ?", createJobDescription());
     assertThat(scheduler.ifJobExist("SampleJob", "SampleService")).isEqualTo(true);
-    scheduler.unregisterJob("SampleJob", "SampleService");
+    assertThat(scheduler.unschedule("SampleJob", "SampleService")).isTrue();
     assertThat(scheduler.ifJobExist("SampleJob", "SampleService")).isEqualTo(false);
+    assertThat(scheduler.unschedule("SampleJob", "SampleService")).isFalse();
   }
 
   @Test
   public void testPauseSchedule() throws Exception {
-    scheduler.registerJob("* * * * * ?", createJobDescription());
-    scheduler.pauseJob("SampleJob", "SampleService");
+    assertThat(scheduler.pauseJobIfPresent("SampleJob", "SampleService")).isFalse();
+    scheduler.scheduleIfAbsent("* * * * * ?", createJobDescription());
+    assertThat(scheduler.pauseJobIfPresent("SampleJob", "SampleService")).isTrue();
     assertThat(scheduler.isJobPaused("SampleJob", "SampleService")).isEqualTo(true);
-    scheduler.resumeJob("SampleJob", "SampleService");
+    assertThat(scheduler.resumeJobIfPresent("SampleJob", "SampleService")).isTrue();
     assertThat(scheduler.isJobPaused("SampleJob", "SampleService")).isEqualTo(false);
 
     // test pausing a paused job
-    scheduler.pauseJob("SampleJob", "SampleService");
-    scheduler.pauseJob("SampleJob", "SampleService");
+    scheduler.pauseJobIfPresent("SampleJob", "SampleService");
+    scheduler.pauseJobIfPresent("SampleJob", "SampleService");
     assertThat(scheduler.isJobPaused("SampleJob", "SampleService")).isEqualTo(true);
     // test resuming a non-paused job
-    scheduler.resumeJob("SampleJob", "SampleService");
-    scheduler.resumeJob("SampleJob", "SampleService");
+    scheduler.resumeJobIfPresent("SampleJob", "SampleService");
+    scheduler.resumeJobIfPresent("SampleJob", "SampleService");
     assertThat(scheduler.isJobPaused("SampleJob", "SampleService")).isEqualTo(false);
   }
 
-  @Ignore("Flaky test, slow too. Don't use Thread.sleep in unit tests.")
-  @Test
-  public void testPauseAndResume() throws Exception {
-    scheduler.registerJob("* * * * * ?", createJobDescription());
-    scheduler.pause();
-    final int count = SampleQuartzJob.COUNT_EXECUTION;
-    Thread.sleep(1500);
-    assertThat(SampleQuartzJob.COUNT_EXECUTION).isEqualTo(count);
-    scheduler.resume();
-    Thread.sleep(1200);
-    assertThat(SampleQuartzJob.COUNT_EXECUTION).isGreaterThan(count);
-  }
-
   private QuartzJobDescription createJobDescription() {
```

Comment:
I'm feeling the naming <JobDescription> might be not appropriate. We may think of a better name.